lib/pages/classes/service/class_provider.dart

```diff
@@ -80,6 +84,17 @@ extension ClassExtension on Class {
       ));
     }
 
+    final lecturerSnap = await FirebaseFirestore.instance
```

Comment:
This behaviour is part of PersonProvider and I think it should stay there, to keep things separate.